src/async/GenericMuxedInterruptDistributor.cc

```diff
@@ -3,27 +3,718 @@
+      {"IVAR", IVAR},     // REMOVE?
+      {"IVEAR", IVEAR},   // REMOVE?
+      {"INVALID_OPTION_CODE", INVALID_OPTION_CODE},
+  });
+
+  /********************************************************************************************************************/
+  inline uint32_t iToMask(const uint32_t ithInterrupt) {
+    return 0x1U << ithInterrupt;
+  }
+
+  /********************************************************************************************************************/
+
+  std::string explainOptCode(GmidOptionCode optCode) {
+    // clang-format off
+      static std::map<GmidOptionCode, std::string> mapOptCode2Message= {
+          {ISR, "Interrupt Status Register"},
+          {IER, "Interrupt Enable Register"},
+          {MER, "Master Enable Register"},
+          {MIE, "Master Interrupt Enable"},
+          {GIE, "Global Interrupt Enable"},
+          {ICR, "Interrupt Clear Register"},
+          {IAR, "Interrupt Acknowledge Register"},
+          {IPR, "Interrupt Pending Register"},
+          {SIE, "Set Interrupt Enable"},
+          {CIE, "Clear Interrupt Enable"},
+          {IMaskR,
+              "Interrupt Mask Register, not to be confused with Interrupt Mode Register"},
+          {IModeR,
+              "Interrupt Mode Register, what AXI INTC v4.1 calls this 'IMR', not to be confused with Interrupt Mask "
+              "Register, which we call 'IMR'"},//REMOVE?
+          {IVR, "Interrupt Vector Register"}, //REMOVE?
+          {ILR, "Interrupt Level Register"},//REMOVE?
+          {IVAR, "Interrupt Vector Address Register"},//REMOVE?
+          {IVEAR, "Interrupt Vector Extended Address Register"},//REMOVE?
+          {INVALID_OPTION_CODE, "No such option code is known"}
+      };
+    // clang-format on
+    return getOptionRegisterStr(optCode) + " (" + mapOptCode2Message[optCode] + ")";
+  }
+
+  /********************************************************************************************************************/
+  GmidOptionCode getOptionRegisterEnum(std::string opt) {
+    auto it = OptionCodeMap.left.find(opt);
+    if(it != OptionCodeMap.left.end()) { // If a valid code
+      return it->second;                 // return the corresponding enum.
+    }
+    else { // invalid code
+      return GmidOptionCode::INVALID_OPTION_CODE;
+    }
+  }
+
+  /********************************************************************************************************************/
+  std::string getOptionRegisterStr(GmidOptionCode optCode) {
+    auto it = OptionCodeMap.right.find(optCode);
+    if(it != OptionCodeMap.right.end()) { // If a valid code
+      return it->second;                  // return the corresponding str.
+    }
+    else { // invalid code
+      return "INVALID_OPTION_CODE";
+    }
+  }
+
+  /********************************************************************************************************************/
+  std::string strSetToStr(const std::set<std::string>& strSet, char delimiter) {
```

Comment:
There is a ticket to move this to the general part of the code. It will become part of the catalogue to have a hierarchical async SubDomain identifier, and we will need convenience and helper functions in the public API for it. I leave it as is for now.